model_analyzer/result/result_manager.py

```diff
@@ -314,15 +341,16 @@ def _compile_measurements(self, result):
 
         passing_measurements = result.passing_measurements()
         failing_measurements = result.failing_measurements()
+
         for (measurements, passes) in [(passing_measurements, True),
                                        (failing_measurements, False)]:
             while measurements:
                 next_best_measurement = heapq.heappop(measurements)
-                self._compile_measurement(model_name=model_name,
-                                          instance_group=instance_group,
-                                          dynamic_batching=dynamic_batching,
-                                          measurement=next_best_measurement,
-                                          passes=passes)
+                self._tabulate_measurement(model_name=model_name,
```

Comment:
By my count, this a quadruple nested loop, if you follow the method call. That doesn't seem algorithmically performant.

Reply:
There is no repeated work. Do you mean inefficient because of simply the overhead of adding an extra nested loop?

Reply:
Right. Is there a way to cache results, for example, from a separate while loop so you don't have to nest. Can you use a sorted list, or hash table? Algorithmically, deeply nested loops are a bad in worst case scenarios. They can also accidentally lead to O(n^2) issues.

Reply:
During collection, each measurement is filled into a nested dictionary 

{model_name : {model_config_name: (model_config, list of measurements)}}

So the data is 3 dimensional inherently. The 4th dimension below is just for convenience to split up passing failing results. 

During report generation we do:
for each model:
    for each config:
          for each of passing/failing:
                for each measurement:
                      ... write to table